dashboard/app/controllers/project_versions_controller.rb

```diff
@@ -16,4 +16,22 @@ def get_token
     headers['csrf-token'] = form_authenticity_token
     return head :ok
   end
+
+  def project_commits
+    user_storage_id, project_id = storage_decrypt_channel_id(params[:channel_id])
+    project_owner = User.find_by(id: user_id_for_storage_id(user_storage_id))
+    return render :not_acceptable unless project_owner
+    return render :forbidden unless can?(:project_commits, ProjectVersion.new, project_owner, project_id)
+    commits = ProjectVersion.where(project_id: project_id).where.not(comment: '').order(created_at: :desc)
+    project_expired_date = 1.year.ago
```

Comment:
It'd be great to have this stored in a const somewhere because we'll also need to use it to tell if the version associated with a code review session is expired

Reply:
And/or create a method on ProjectVersion instance- `is_expired`, could be helpful

Reply:
Happy to create a method. Where do you think the constant should go? I avoided creating one because I wasn't sure where else needed it

There is an edge case where the current version of a project will never expire, no matter how old it is, but I don't think we have an easy way of checking if a version is the "current" version.

Reply:
Oh that's an interesting point. I think for the purposes of this feature if we treat all versions older than a year as expired, that's probably ok. I don't think this is currently the kind of feature where students will revisit it more than a year later.

It'll also be needed in our CodeReviewRequests model. What do you think about adding a new helper to `dashboard/app/helpers` called `project_version_helper` which could include either the constant to import or a helper method which takes the date and returns a boolean?

Reply:
Done in [69b43ca](https://github.com/code-dot-org/code-dot-org/pull/46139/commits/69b43caab63427c7f823986ba49e3a35f6bad5c1)!